docs/madr/decisions/020-kds-redesign.md

```diff
@@ -0,0 +1,179 @@
+# KDS Redesign
+
+* Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/4354
+
+## Context and Problem Statement
+
+The current implementation of KDS is quite complex, and there may be some bugs, such as the one found at https://github.com/kumahq/kuma/pull/5373, which was very difficult to identify and resolve. The use of a bi-directional stream for configuration exchange between the zone and the global can also be challenging to implement. Additionally, KDS operates as a State of The World, meaning that whenever an item in the resource type changes, the entire state is sent. While this doesn't appear to be a problem when synchronizing data from Global to Zone, it may be less efficient when there are many Dataplane changes. There are several improvements we want to make in the future such as:
+* working with different older versions of the control plane on the zone,
+* creating separate gRPC services for synchronization between the zone and global,
+* supporting multiple control plane tenants,
+* prioritizing sync of operational over informational resources,
+* changing `default` namespace for sync to global to system one (`kuma-system`), 
+* and sending only changes instead of the state of the world when synchronizing data from the zone to the global.
+
+## Considered Options
```

Comment:
Shouldn't this contain some number of specific redesigns as opposed to "redesign or don't redesign"?

Reply:
Added